packages/backend/src/services/fetch-instance-metadata.ts

```diff
@@ -156,7 +156,7 @@ async function fetchFaviconUrl(instance: Instance, doc: DOMWindow['document'] |
 	const url = 'https://' + instance.host;
 
 	if (doc) {
-		const href = doc.querySelector('link[rel="icon"]')?.getAttribute('href');
+		const href = doc.querySelector('link[rel="icon"], link[rel^="icon "], link[rel$=" icon"], link[rel*=" icon "]')?.href;
```

Comment:
```suggestion
		const href = Array.from(doc.getElementsByTagName('link')).filter(link => link.relList.contains('icon'))[0]?.href;
```

@acid-chicken 曰くrelListを使うのがスマートらしいのでこう書くべきなのかなと

Reply:
But it's the same behaviour? My idea was that the CSS selector will be more optimized because it is implemented in Node directly, instead of doing the same thing in JavaScript by ourselves.

Reply:
私はむしろquerySelectorが遅いものと思っていました。エビデンスはありませんが。

Reply:
fyi: querySelector impl https://github.com/dperini/nwsapi/blob/master/src/nwsapi.js

Reply:
つまり…どういうことなんだ？(

Reply:
使っているのは Node.js の API ではなくて npm の jsdom で、その jsdom はこのライブラリを参照しているので、別にネイティブ実装ではなくどの道 JS のランタイム上で動いている点だけ補足したつもり（どちらがパフォーマンスいいかは自信を持って回答できるものがない）

Reply:
ここは速度が要るような場面ではなさそうなので、可読性重視でいきたい

Reply:
filter()[0]じゃなくてfindでいいことに気づいたのでsuggestを書き換えた(

Reply:
updated to the latest suggestion

Reply:
> filter()[0]じゃなくてfindでいいことに気づいたのでsuggestを書き換えた

これ、一番後ろを参照すべき